cadence/contracts/services/AxelarGovernanceService.cdc

```diff
@@ -0,0 +1,296 @@
+    }
+
+    //Execute Scheduled Proposal
+    access(all) fun executeProposal(proposedCode: String, target: Address){
+        let proposalHash: String = String.encodeHex(self.createProposalHash(proposedCode: proposedCode, target: target))
+        //check for time left in propsoal
+        if(self.proposals[proposalHash]?.getTimeToExecute()! <= UInt64(getCurrentBlock().timestamp)){
+            //Execute Proposal
+            self.proposals[proposalHash]?.execute()
+
+            emit ProposalExecuted(
+                proposalHash: proposalHash,
+                target: target,
+                proposedCode: proposedCode,
+                timestamp: UInt64(getCurrentBlock().timestamp)
+            )
+
+            destroy <- self.proposals.remove(key: proposalHash)
+        } else { 
+            panic("ProposalNotReady")
+        }
+    }
+
+    access(all) resource ExecutabeResource: AxelarGateway.Executable, AxelarGateway.SenderIdentity{
+        access(all) fun executeApp(commandResource: &AxelarGateway.CGPCommand, sourceChain: String, sourceAddress: String, payload: [UInt8]): AxelarGateway.ExecutionStatus{
+            if (sourceChain != AxelarGovernanceService.governanceChain || sourceAddress != AxelarGovernanceService.governanceAddress){
+                panic("Not Governance")
+            }
+
+
+            // let commandSelector = payload[0]
+            // let target = Address.fromBytes(payload[1])
+            // let proposedCode = String.fromUTF8(payload[2])!
+            // let timeToExecute = UInt64.fromString(String.fromUTF8(payload[3])!)!
+            // let contractName = String.encodeHex(payload[4])
+
+            //defining as constants untill abiencode/decode is included
+            let commandSelector = AxelarGovernanceService.SELECTOR_SCHEDULE_PROPOSAL
+            let target: Address= AxelarGovernanceService.gateway
+            let proposedCode = String.fromUTF8(payload)!
+            let timeToExecute = 0 as UInt64
+            let contractName = "AxelarGateway"
+
+            AxelarGovernanceService._processCommand(commandSelector: commandSelector, proposedCode: proposedCode, target: target, timeToExecute: timeToExecute, contractName: contractName)
+            return AxelarGateway.ExecutionStatus(
+                isExecuted: true,
+                statusCode: 0,
+                errorMessage: ""
+            )
+        }
+    }
+
+    //Process Command coming from Gateway
+    access(contract) fun _processCommand(commandSelector: [UInt8] ,proposedCode: String, target: Address, timeToExecute: UInt64, contractName: String){
+        let proposalHash = String.encodeHex(self.createProposalHash(proposedCode: proposedCode, target: target))
+        if (commandSelector == self.SELECTOR_SCHEDULE_PROPOSAL){
+            self.proposals[proposalHash] <-! create Proposal(id: proposalHash, proposedCode: proposedCode, target:target, contractName: contractName, timeToExecute: timeToExecute)
```

Comment:
In the event of a collision - e.g. I've submitted a proposal with the same code and target - how do we want to handle that case? An upsert would remove and destroy any existing proposals or we could panic and revert if one already exists